python/ray/data/dataset.py

```diff
@@ -323,7 +323,7 @@ def map_batches(
         self,
         fn: BatchUDF,
         *,
-        batch_size: Optional[int] = DEFAULT_BATCH_SIZE,
+        batch_size: Optional[Union[int, Literal["default"]]] = "default",
```

Comment:
Can you update the documentation of `batch_size` below as well?

Reply:
@jianoaix - could you suggest the specific documentation to be updated? It already mentions `Defaults to 4096`. Feel quite clear for end users.

Reply:
So this introduced a new default value "default", right? We need to document what does that do to the semantics.

Reply:
Thought `Defaults to 4096` is clear, but updated anyway. Could you help double check?